src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,75 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (!isMacintosh) {
+			// We spawn process.execPath directly
+			child = spawn(process.execPath, argv.slice(2), options);
+		} else {
+			// On mac, we spawn using the open command to obtain behavior
+			// similar to if the app was launched from the dock
+			// https://github.com/microsoft/vscode/issues/102975
+			const openArgs: string[] = ['-n'];
+
+			if (verbose || hasReadStdinArg || args.wait) {
+				openArgs.push('-W');
```

Comment:
```suggestion
				openArgs.push('--wait-apps');
```

Reply:
Makes it a bit clearer what the option does. Maybe we use the more verbose form for all?